test/python/transpiler/test_template_matching.py

```diff
@@ -353,9 +354,24 @@ def template():
         pass_ = TemplateOptimization(template_list=[template()])
         circuit_out = PassManager(pass_).run(circuit_in)
 
-        # This template will not fully match as long as gates with parameters do not
-        # commute with any other gates in the DAG dependency.
-        self.assertEqual(circuit_out.count_ops().get("cx", 0), 2)
+        self.assertEqual(circuit_out.count_ops().get("cx", 0), 0)
+
+    def test_unbound_parameters_in_rzx_template(self):
+        """
+        Test that rzx template ('zz2') functions correctly for a simple
+        circuit with an unbound ParameterExpression.
+        """
+
+        phi = Parameter("$\\phi$")
+        circuit_in = QuantumCircuit(2)
+        circuit_in.cx(0, 1)
+        circuit_in.p(2 * phi, 1)
+        circuit_in.cx(0, 1)
+
+        pass_ = TemplateOptimization(**rzx_templates(["zz2"]))
+        circuit_out = PassManager(pass_).run(circuit_in)
+
+        self.assertEqual(circuit_out.count_ops().get("cx", 0), 0)
```

Comment:
Is there an exact expected output of this circuit?  It'd be good to really make the assertions as tight as possible, to test that the optimization has completely succeeded - at present, if it just returned an empty circuit it'd still pass this test.

Also, can we add some more complex tests when parameters are parameter expressions with multiple different symbols in them as well?

Reply:
Fixed

Reply:
This isn't actually fixed - it's got the same assertions as before.  A good test would be to construct the exact output circuit you expect, and then do `self.assertEqual(circuit_out, expected)`.

Reply:
Fixed this time!